src/main/java/org/checkerframework/languageserver/CFTextDocumentService.java

```diff
@@ -184,20 +197,177 @@ public void didSave(DidSaveTextDocumentParams params) {
     public void publish(Map<String, List<javax.tools.Diagnostic<?>>> result) {
         for (Map.Entry<String, List<javax.tools.Diagnostic<?>>> entry : result.entrySet()) {
             List<Diagnostic> diagnostics = new ArrayList<>();
+            Map<String, List<CheckerTypeKind>> uniqueTypeInfo =
+                    processDiagnosticString(entry, diagnostics);
+            publishTypeMessageWithFilter(entry, uniqueTypeInfo);
+            server.publishDiagnostics(new PublishDiagnosticsParams(entry.getKey(), diagnostics));
+        }
+    }
+
+    /**
+     * Process Diagnostic strings to separate type message and error message.
+     *
+     * @param entry The Diagnostic entry for getting diagnostic message
+     * @param diagnostics A list for storing checkerframework issued errors
+     * @return TypeMessage processed for non-verbose output
+     */
+    private Map<String, List<CheckerTypeKind>> processDiagnosticString(
+            Map.Entry<String, List<javax.tools.Diagnostic<?>>> entry,
+            List<Diagnostic> diagnostics) {
+        Map<String, List<CheckerTypeKind>> TypeMessage = new HashMap<>();
+        for (javax.tools.Diagnostic<?> diagnostic : entry.getValue()) {
+            String message = diagnostic.getMessage(Locale.getDefault());
+            if (message != null && message.contains("lsp.type.information")) {
+                String checker = getChecker(message);
+                String kind = getKind(message);
+                String type = getType(message);
+                String positionInfo = getPosition(message);
+                if (positionInfo != null) {
+                    if (!TypeMessage.containsKey(positionInfo)) {
+                        List<CheckerTypeKind> checkerTypeKinds = new ArrayList<>();
+                        Map<String, String> kindType = new HashMap<>();
+                        kindType.put(type, kind);
+                        checkerTypeKinds.add(new CheckerTypeKind(checker, kindType));
+                        TypeMessage.put(positionInfo, checkerTypeKinds);
+                    } else {
+                        List<CheckerTypeKind> checkerTypeKinds = TypeMessage.get(positionInfo);
+                        boolean foundChecker = false;
+                        for (CheckerTypeKind checkerTypeKind : checkerTypeKinds) {
+                            if (checkerTypeKind.getCheckername().equals(checker)) {
+                                foundChecker = true;
+                                if (!checkerTypeKind.getTypeKind().containsKey(type)) {
```

Comment:
These methods have a very deep nesting level. Can you try to split off smaller helper methods? Or at least add some comments to make the flow understandable?

Reply:
already refactored and added comment.